### PR TITLE
agent: update resources list with the right device major-minor number

### LINKED ISSUE #351: devices cgrup doesn't reflect the right list of devices
Since host device's major-minor numbers are mapped to guest numbers, the list of resources must be updated to apply devices constraints correctly.

more info see https://github.com/kata-containers/runtime/issues/677

### PR DESCRIPTION
**Pull request by devimc**
Host device's major-minor numbers are mapped to guest major-minor numbers,
for example in the host the major-minor number for /dev/loop0p1 is 259:1,
when it's attached to the VM now the major-minor number is 8:0, these
conversion must be reflected in devices and resources lists, the first list
is used to mount the device in the container and the second one is to update
the devices cgroup.

fixes #351

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Looks good, but please can you add a unit test?

**Comment by amshinde** [comment]
lgtm
@devimc Can you add some documentation in for the device cgroups feature?

**Comment by devimc** [comment]
@amshinde do you mean more comments in the code?

**Comment by devimc** [comment]
@jodh-intel done, I have added a unit test for whole function, thanks

File: device_test.go [hunk 1]
```diff
@@ -373,3 +373,73 @@ func TestAddDevice(t *testing.T) {
 		}
 	}
 }
+
+func TestUpdateSpecDeviceList(t *testing.T) {
```
Comment (jodh-intel) [inline device_test.go]: Very nice! Thanks @devimc! :smile:

**Pull request merged**